controller/users.go

```diff
@@ -254,6 +268,40 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 	})
 }
 
+func isUsernameUnique(appl application.Application, username string, identity account.Identity) (bool, error) {
+	usersWithSameUserName, err := appl.Identities().Query(account.IdentityFilterByUsername(username))
```

Comment:
Add a filter for KC identities only.
Remote identities (from githbut, jira, etc) can match our KC identity usernames. That's OK. We care about KC identities duplications only when checking usernames.